Synapse 1.48.0 (#1130)
matrix-org/matrix.org: gatsby/content/blog/2021/11/2021-11-30-synapse-1.48.0-released.mdx

---
date: '2021-11-30'
title: Synapse 1.48.0 released
categories:
- Releases
author: Brendan Abolivier
---

[Synapse 1.48.0](https://github.com/matrix-org/synapse/releases/tag/v1.48.0) is out now!

> 

Reviewer: ```suggestion
> __NOTE:__ Synapse 1.49, due out on December 14th, will be the **last release of Synapse to support Python 3.6 or PostgreSQL 9.6** per our [platform dependency deprecation policy](https://matrix-org.github.io/synapse/v1.48/deprecation_policy.html). Accordingly, we will remove support for Ubuntu 18.04 LTS (Bionic) at the same date, as it ships with Python 3.6.

```

## Password resets and identity servers

This release removes the long-deprecated `trust_identity_server_for_password_resets` configuration option. This option was initially deprecated in [Synapse 1.4.0](https://github.com/matrix-org/synapse/releases/tag/v1.4.0) back in October 2019.

Admins of servers still using this configuration option will need to update their Synapse configuration to send password resets through an SMTP server directly rather than relying on identity servers to send them on their behalf.

> ## New [admin APIs](https://matrix-org.github.io/synapse/latest/usage/administration/admin_api/index.html) and more compatibility with the Matrix specification

Reviewer: ```suggestion
## New [admin APIs](https://matrix-org.github.io/synapse/latest/usage/administration/admin_api/index.html) and improved alignment with Matrix 1.1
```

This release also introduces a handful of new admin APIs, allowing administrators to [un-shadow-ban users](https://matrix-org.github.io/synapse/latest/admin_api/user_admin_api.html#controlling-whether-a-user-is-shadow-banned), [block a room](https://matrix-org.github.io/synapse/latest/admin_api/rooms.html#block-room-api), and [run specific background updates](https://matrix-org.github.io/synapse/latest/usage/administration/admin_api/background_updates.html#run) (but we'll talk about this last one a bit later on). The [delete room API](https://matrix-org.github.io/synapse/latest/admin_api/rooms.html#delete-room-api) has also been updated to be able to run in the background or to block a room pre-emptively, even if the server doesn't know about it yet.

> This release also brings Synapse closer to compatiblity with [Matrix v1.1](https://matrix.org/blog/2021/11/09/matrix-v-1-1-release) by adding support for the `/_matrix/client/v3` and `/_matrix/media/v3` API from the latest version of the specification.

Reviewer 2: ```suggestion
This release also brings Synapse closer to compatiblity with [Matrix v1.1](https://matrix.org/blog/2021/11/09/matrix-v-1-1-release) by adding support for the `/_matrix/client/v3` and `/_matrix/media/v3` APIs from the latest version of the specification.
```

Lately the Synapse team has been doing some work to improve the performance of these background updates. More specifically, this release includes a [performance fix](https://github.com/matrix-org/synapse/pull/11421) for a background update introduced in Synapse 1.47.0, as well as a [new admin API](https://matrix-org.github.io/synapse/latest/usage/administration/admin_api/background_updates.html#run) to let admins rerun specific updates.

Future Synapse updates will also include [module capabilities](https://github.com/matrix-org/synapse/pull/11306) and [more configuration options](https://github.com/matrix-org/synapse/issues/11260) for controlling background updates.

## Everything else

This release also includes some improved support of [MSC3440](https://github.com/matrix-org/matrix-doc/pull/3440) to help threading. It also adds support for the stable identifiers from [MSC2778](https://github.com/matrix-org/matrix-doc/pull/2778), bringing Synapse closer to supporting end-to-end (or end-to-bridge) encryption support for application services.

> 

Reviewer: ```suggestion

We also now publish a Docker image, `matrixdotorg/synapse:develop`, which tracks the development head of Synapse.

```

---
date: '2021-11-30'
title: Synapse 1.48.0 released
categories:
- Releases
author: Brendan Abolivier
---

> [Synapse 1.48.0](https://github.com/matrix-org/synapse/releases/tag/v1.48.0) is out now!
> > __NOTE:__ Synapse 1.49, due out on December 14th, will be the **last release of Synapse to support Python 3.6 or PostgreSQL 9.6** per our [platform dependency deprecation policy](https://matrix-org.github.io/synapse/v1.48/deprecation_policy.html). Accordingly, we will remove support for Ubuntu 18.04 LTS (Bionic) at the same date, as it ships with Python 3.6.

Author: ```suggestion
[Synapse 1.48.0](https://github.com/matrix-org/synapse/releases/tag/v1.48.0) is out now!

> __NOTE:__ Synapse 1.49, due out on December 14th, will be the **last release of Synapse to support Python 3.6 or PostgreSQL 9.6** per our [platform dependency deprecation policy](https://matrix-org.github.io/synapse/v1.48/deprecation_policy.html). Accordingly, we will remove support for Ubuntu 18.04 LTS (Bionic) at the same date, as it ships with Python 3.6.
```

> > __NOTE:__ Synapse 1.49, due out on December 14th, will be the **last release of Synapse to support Python 3.6 or PostgreSQL 9.6** per our [platform dependency deprecation policy](https://matrix-org.github.io/synapse/v1.48/deprecation_policy.html). Accordingly, we will remove support for Ubuntu 18.04 LTS (Bionic) at the same date, as it ships with Python 3.6.

Reviewer: Hm. GitHub dropped a newline here. Let's add it back.

```suggestion

> __NOTE:__ Synapse 1.49, due out on December 14th, will be the **last release of Synapse to support Python 3.6 or PostgreSQL 9.6** per our [platform dependency deprecation policy](https://matrix-org.github.io/synapse/v1.48/deprecation_policy.html). Accordingly, we will remove support for Ubuntu 18.04 LTS (Bionic) at the same date, as it ships with Python 3.6.
```

Author: Way ahead of you :stuck_out_tongue: https://github.com/matrix-org/matrix.org/pull/1130#discussion_r760370938
